{{cookiecutter.project_slug}}/.github/workflows/proxy.yml

```diff
@@ -0,0 +1,97 @@
+name: Deploy Proxy
+
+on:
+  push:
+    branches:
+      - sandbox
+      - develop
+      - staging
+      - testing
+    paths:
+      - .github/workflows/proxy.yml
+      - 'proxy/**'
+  workflow_dispatch:
+    branches:
+      - temp
+      - demo
+      - sandbox
+      - testing
+      - develop
+      - prod
+
+permissions:
+  packages: write
+  repository-projects: read
+  id-token: write
+  contents: read
+
+
+jobs:
+  deploy_proxy:
+
+    runs-on: ubuntu-latest
+
+    env:
+      CF_SPACE: ${{ github.ref_name }}
+
+    strategy:
+      matrix:
+        app: [{{cookiecutter.package_name}}]
+
+    steps:
+
+      - uses: actions/checkout@v3
+
+      - name: Download CF CLI
+        working-directory: ./proxy
+        run: curl -L "https://paas-s3-broker-prod-lon-7c642b9d-e09c-4dec-9fce-0e1e82b051e6.s3.amazonaws.com/cfcli/cf7-cli_7.6.0_linux_x86-64.tgz" | tar -zx
+
+      - name: CF login
+        working-directory: ./proxy
+        run: ./cf login -a api.london.cloud.service.gov.uk -u "${{ secrets.CF_USER }}" -p "${{ secrets.CF_PASSWORD }}" -o co-i-ai -s "$CF_SPACE"
+
+      - name: Get IPs
+        id: get_ips
+        working-directory: ./proxy
+        run: echo "::set-output name=ips::$(grep -o '^[^#]*' whitelisted_ips.txt | tr '\n' ' ')"
+
+      - name: CF push proxy apps non-prod
+        if: ${{ github.ref_name != 'prod' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_ACTOR: ${{ github.actor }}
+          ROUTE_SERVICE_APP_NAME: ${{ matrix.app }}-proxy-${{ github.ref_name }}
+          ROUTE_SERVICE_NAME: ${{ matrix.app }}-route-${{ github.ref_name }}
+          PROTECTED_APP_NAME: ${{ matrix.app }}-${{ github.ref_name }}
+          ALLOWED_IPS: ${{ steps.get_ips.outputs.ips }}
+          HTPASSWD: ${{ secrets.HTPASSWD }}
+        working-directory: ./proxy
+        run: |
+            ./proxy-deploy.sh
+
+      - name: CF push proxy apps prod
+        if: ${{ github.ref_name == 'prod' }}
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          GITHUB_ACTOR: ${{ github.actor }}
+          ROUTE_SERVICE_APP_NAME: ${{ matrix.app }}-proxy
+          ROUTE_SERVICE_NAME: ${{ matrix.app }}-route
+          PROTECTED_APP_NAME: ${{ matrix.app }}
+          ALLOWED_IPS: ${{ steps.get_ips.outputs.ips }}
+          HTPASSWD: ${{ secrets.HTPASSWD }}
+        working-directory: ./proxy
+        run: |
+            ./proxy-deploy.sh
+
+      - uses: actions/checkout@v2
```

Comment:
Let's get rid of the Slack Notifications - no-one uses them, and some of us don't understand them.